clients/command/src/main/java/org/eclipse/hono/client/command/ProtocolAdapterCommandConsumerFactoryImpl.java

```diff
@@ -72,19 +87,34 @@ public class ProtocolAdapterCommandConsumerFactoryImpl implements ProtocolAdapte
      */
     public ProtocolAdapterCommandConsumerFactoryImpl(final CommandRouterClient commandRouterClient, final String adapterName) {
         this.commandRouterClient = Objects.requireNonNull(commandRouterClient);
-        Objects.requireNonNull(adapterName);
+        this.adapterName = Objects.requireNonNull(adapterName);
 
-        this.adapterInstanceId = CommandRoutingUtil.getNewAdapterInstanceId(adapterName,
-                ADAPTER_INSTANCE_ID_COUNTER.getAndIncrement());
+        this.adapterInstanceIdCounterValue = ADAPTER_INSTANCE_ID_COUNTER.getAndIncrement();
         if (commandRouterClient instanceof ConnectionLifecycle<?>) {
             ((ConnectionLifecycle<?>) commandRouterClient).addReconnectListener(con -> reenableCommandRouting());
         }
     }
 
+    /**
+     * Sets the vert.x context to run blocking code on.
+     * <p>
+     * If not set explicitly, the context is determined during startup.
+     *
+     * @param context The context.
+     * @throws NullPointerException if context is {@code null}.
+     */
+    void setContext(final Context context) {
```

Comment:
is there a particular reason why you do not want to pass in a Vertx instance to the constructor as we do in most other services?

Reply:
Right, that is the better solution. I've changed it now.